board/designa/snappermx6/snappermx6.c

```diff
@@ -39,60 +40,75 @@ static iomux_v3_cfg_t const uart5_pads[] = {
-	MX6_PAD_ENET_REF_CLK__ENET_TX_CLK    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_TX_EN__ENET_TX_EN    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_TXD0__ENET_TX_DATA0	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_TXD1__ENET_TX_DATA1   	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_ROW2__ENET_TX_DATA2		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_ROW0__ENET_TX_DATA3		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_GPIO_18__ENET_RX_CLK		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_CRS_DV__ENET_RX_EN    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_RXD0__ENET_RX_DATA0	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_RXD1__ENET_RX_DATA1	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_COL2__ENET_RX_DATA2		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_COL0__ENET_RX_DATA3		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_RX_ER__ENET_RX_ER    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_ROW1__ENET_COL		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_COL3__ENET_CRS		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-
-	/* PHY reset */
-	MX6_PAD_KEY_COL1__GPIO4_IO08		| MUX_PAD_CTRL(NO_PAD_CTRL),
-};
-
-#ifdef CONFIG_MXC_SPI
-static iomux_v3_cfg_t const ecspi1_pads[] = {
-	MX6_PAD_EIM_EB2__GPIO2_IO30 | MUX_PAD_CTRL(SPI_PAD_CTRL),
-	MX6_PAD_EIM_D18__ECSPI1_MOSI | MUX_PAD_CTRL(SPI_PAD_CTRL),
-	MX6_PAD_EIM_D17__ECSPI1_MISO | MUX_PAD_CTRL(SPI_PAD_CTRL),
-	MX6_PAD_EIM_D16__ECSPI1_SCLK | MUX_PAD_CTRL(SPI_PAD_CTRL),
-};
-
-int board_spi_cs_gpio(unsigned bus, unsigned cs)
+static int salmon_version_read(void)
 {
-	return (bus == 0 && cs == 0) ? (IMX_GPIO_NR(2, 30)) : -1;
+	int retval = 0;
+	// We need to do this using a low-level read as this is used prior to the device
+	// tree being initialised, so the dm_ functions will not work
+	// We're on GPIO bank 2, pins 9 & 10
+
+	uint32_t gpio_state = readl(GPIO2_BASE_ADDR);
+
+	if (gpio_state & (1 << 10))
+		retval |= 1;
+	if (gpio_state & (1 << 9))
+		retval |= 2;
+
+	// Convert from the pull-up states to more useful version numbers
+	// Mainboard rev2 uses in-CPU pulls up, so we get all 0x3
+	// Mainbaord rev3 uses a pull-down on VERSION0, so we get 0x2
+	// Future mainboards not yet defined
+	if (retval == 0x3) {
+		return 2;
+	} else if (retval == 0x2) {
+		return 3;
+	} else {
+		printf("WARNING: Unknown mainboard strapping state 0x%x\n", retval);
+		return 0;
+	}
 }
 
-static void setup_spi(void)
+#ifdef CONFIG_FEC_MXC
+static int setup_fec_clock(void)
 {
-	SETUP_IOMUX_PADS(ecspi1_pads);
+	struct iomuxc *iomuxc_regs = (struct iomuxc *)IOMUXC_BASE_ADDR;
+
+	/* set gpr1[21] to select anatop clock */
+	setbits_le32(&iomuxc_regs->gpr[1], IOMUXC_GPR1_ENET_CLK_SEL_MASK);
+	iomuxc_set_rgmii_io_voltage(DDR_SEL_1P5V_IO);
+	enable_fec_anatop_clock(0, ENET_125MHZ);
+	enable_enet_clk(1);
+	return 0;
 }
-#endif
 
-static void setup_iomux_enet(void)
+static void salmon_enet_init(void)
 {
 	int reset_gpio = IMX_GPIO_NR(4, 8);
-	SETUP_IOMUX_PADS(enet_pads);
-
-	/* Reset the 88e6061 PHY on the Salmon carrier board */
+	int version = salmon_version_read();
```

Comment:
This should probably be a sysinfo driver

Reply:
I'm not familiar with this. Do you mean like `omap3_sysinfo`, using `checkboard` etc...? That doesn't seem to be a generic piece of infrastructure?

Reply:
It's in sysinfo.h - basically a driver that returns system info

Reply:
sysinfo.h appears to be x86/arch-coreboot specific?